From: Jitendra Kalsaria <jitendra.kalsaria@qlogic.com>
Date: Wed, 18 Aug 2010 19:04:00 -0700
Subject: [PATCH 06/14] qlge: fix a eeh handler to not add a pending timer.
Git-commit: 7ae80abdba0644e12ac17da567a2db1efc1bf8a8
Patch-mainline: v2.6.35-rc5

	On some ocasions the function qlge_io_resume() tries to add a
	pending timer, which causes the system to hit the BUG() on
	mod_timer() function.

	This patch removes the timer during the EEH recovery.

Signed-off-by: Jitendra Kalsaria <jitendra.kalsaria@qlogic.com>
Acked-by: Jeff Mahoney <jeffm@suse.com>
---
 drivers/net/qlge/qlge_main.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/drivers/net/qlge/qlge_main.c b/drivers/net/qlge/qlge_main.c
index dbb4acb..041dc7d 100644
--- a/drivers/net/qlge/qlge_main.c
+++ b/drivers/net/qlge/qlge_main.c
@@ -4700,6 +4700,8 @@ static void ql_eeh_close(struct net_device *ndev)
 		netif_carrier_off(ndev);
 		netif_stop_queue(ndev);
 	}
+	/* Disabling the timer */
+	del_timer_sync(&qdev->timer);
 
 	if (test_bit(QL_ADAPTER_UP, &qdev->flags))
 		cancel_delayed_work_sync(&qdev->asic_reset_work);
-- 
1.7.6.rc0.12.g2c6b5

